### PR TITLE
better docs: osproc

### PR DESCRIPTION
**Pull request by narimiran**
Other than more informative docs, fixes typo in the name `poDemon` to `poDaemon`.

### CODE REVIEW

**Comment by sg-james** [comment]
This typo is a breaking change. 
I'd prefer to see it in a separate PR with a change log entry

**Comment by narimiran** [comment]
> This typo is a breaking change.

True.

> change log entry

Agreed, I should add it.

> a separate PR

I'll see what others say.

**Comment by GULPF** [comment]
It's possible to deprecate individual enum values, so the old spelling can still be kept around for backwards compatibility.

File: lib/pure/osproc.nim [hunk 1]
```diff
@@ -26,20 +31,21 @@ when defined(linux):
...
-    poInteractive,       ## optimize the buffer handling for responsiveness for
+    poStdErrToStdOut,    ## Merge stdout and stderr to the stdout stream.
+    poParentStreams,     ## Use the parent's streams.
+    poInteractive,       ## Optimize the buffer handling for responsiveness for
                          ## UI applications. Currently this only affects
                          ## Windows: Named pipes are used so that you can peek
                          ## at the process' output streams.
-    poDemon              ## Windows: The program creates no Window.
+    poDaemon             ## Windows: The program creates no Window.
                          ## Unix: Start the program as a demon. This is still
```
Comment (GULPF) [inline lib/pure/osproc.nim]: Doc comment is also misspelled, demon -> daemon

**Comment by narimiran** [comment]
> It's possible to deprecate individual enum values, so the old spelling can still be kept around for backwards compatibility.

Can you tell me how? I've tried this:
```nim
    poDemon, {.deprecated: "use poDaemon".},
    poDaemon             ## Windows: The program creates no Window.
                         ## Unix: Start the program as a daemon. This is still
                         ## work in progress!
```

but I get `Error: annotation to deprecated not supported here`.

**Comment by Araq** [comment]
Do it this way: Rename it to ``poDaemon``, introduce ``const poDemon {.deprecated: "blah".} = poDaemon``

**Comment by Araq** [comment]
I'm merging this now but this is far from complete: It should link to a document of how to accomplish common tasks like redirecting streams, etc.

**Pull request merged**